MM-11382 Remove special handling of PluginSettings when loading config

hmhealey opened the pull request:
Because of some issues reading settings keys containing periods, this code used to pave over the plugin settings with the contents of `MM_PLUGINSETTINGS`. That was preventing the user from specifying any single plugin setting from the environment, even if `MM_PLUGINSETTINGS` is not being used. This makes sure that setting is actually in use before paving over the PluginSettings field.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-11382

#### Checklist
- Added or updated unit tests (required for all new features)

lieut-data commented on utils/config.go:
```diff
@@ -250,8 +250,10 @@ func ReadConfig(r io.Reader, allowEnvironmentOverrides bool) (*model.Config, map
 		// https://github.com/spf13/viper/issues/348
-		config.PluginSettings = model.PluginSettings{}
-		unmarshalErr = v.UnmarshalKey("pluginsettings", &config.PluginSettings)
+		if _, ok := os.LookupEnv("MM_PLUGINSETTINGS"); ok {
```
Maybe I'm missing something, but the linked issues talk about periods in the key names. Isn't the `UnmarshalKey` critical to ensure we parse the plugin settings data structure even if the plugin ids contain periods? And that we should be doing this independent of whether or not settings are coming from the environment?

hmhealey replied:
I think the purpose of the previous code was to explicitly call `v.UnmarshalKey` to set the entire PluginSettings object using `MM_PLUGINSETTINGS` because you can't use an environment variable like `MM_PLUGINSETTINGS_PLUGINS_COM.EXAMPLE.PLUGIN` because of the linked issues.

hmhealey commented:
@lieut-data From what I can tell, that code isn't actually needed any more because we changed the definition of `PluginSettings.Plugins` from `map[string]interface{}` to `map[string]map[string]interface{}` in [this PR](https://github.com/mattermost/mattermost-server/pull/9066). I added a test case for the problem case that fails with the old type declaration for that field and passes with the new one.

lieut-data commented on utils/config_test.go:
```diff
@@ -37,6 +37,41 @@ func TestReadConfig(t *testing.T) {
+			"Directory": "/temp/mattermost-plugins",
+			"Plugins": {
+				"com.example.plugin": {
+					"number": 1,
```
Given the weirdness of the original issue, can you add another configuration that, itself, contains a period and maybe has a more complex type? e.g. 
```go
"struct.important": {
    "nested.number": 1, 
    "nested.string": "abc", 
    "nested.boolean": false
}
```

Just to ensure we don't have any more weirdness going forward :)
